src/plot_api/plot_api.js

```diff
@@ -2234,6 +2243,9 @@ exports.react = function(gd, data, layout, config) {
             gd._context = undefined;
             setPlotContext(gd, config);
             configChanged = diffConfig(oldConfig, gd._context);
+
+            // remove responsive handler
+            Lib.clearResponsive(gd);
```

Comment:
hmm actually there's another way to turn off `config.responsive` on a `div`, by calling `newPlot` again. I think the removal belongs in `Plotly.plot` instead, next to where you add the handler:
```js
if(gd._context.responsive) {
    if(!gd. _responsiveChartHandler) { ...attach }
}
else {
    Lib.clearResponsive(gd);
}
```
Then it would handle both cases.